langchain/prompts/chat.py

```diff
@@ -0,0 +1,28 @@
+from abc import ABC
+from typing import Any, List, Tuple
+
+from langchain.prompts.base import BasePromptTemplate
+from langchain.schema import ChatMessage
+
+
+class ChatPromptTemplate(BasePromptTemplate, ABC):
+    input_variables: List[str]
```

Comment:
Should this be a getter that does union of input_variables of all prompts in `messages`?

Reply:
ill leave as now, but add a class constructor to create from all prompts in messages

Reply:
ya class constructor would be good, added one in ts here: https://github.com/hwchase17/langchainjs/pull/197/files